src/chat/MarkAsReadButton.js

```diff
@@ -55,6 +62,23 @@ export default function MarkAsReadButton(props: Props): Node {
     }
   }, [auth, narrow, streams]);
 
+  const markPmAsRead = useCallback(() => {
+    const userIds = userIdsOfPmNarrow(narrow);
+    let messageIds = undefined;
+    const key = pmUnreadsKeyFromPmKeyIds(userIds, ownUserId);
+    if (userIds.length > 1) {
+      messageIds = unreadHuddles.find(x => x.user_ids_string === key)?.unread_message_ids ?? [];
+    } else {
+      messageIds = unreadPms.find(x => x.sender_id.toString() === key)?.unread_message_ids ?? [];
+    }
+
+    // The message IDs come from our unread-messages data, which is
+    //   initialized with "only" the most recent 50K unread messages. So
+    //   we'll occasionally, but rarely, miss some messages here; see #5156.
+    // TODO(?): Should we use `queueMarkAsRead` here?
```

Comment:
No need for that, I think; the job of that queue is when you're scrolling and have lots of frequent little batches to mark as read.